lib/upgrader.mli

```diff
@@ -22,21 +28,28 @@ val user_fns_module_name : string -> string
 val convert : string
 
 val name_upgrader_module
-  :  old_file_version:int
-  -> new_file_version:int
+  :  old_file_version:Version.t
+  -> new_file_version:Version.t
   -> string
 
-val name_t_module : string -> int -> string
+val name_t_module : string -> Version.t -> string
 
-val name_impl_module : Config.impl_kind -> string -> int -> string
+val name_impl_module : Config.impl_kind -> string -> Version.t -> string
 
 val load_sort_map
-  :  string
-  -> ((string * string list * (string * (string * string option) list) list)
-     * Ast.NoLoc.TypeExpr.t)
-     list
-     * ((string * Ast.NoLoc.ModuleItem.type_param * Ast.NoLoc.annot)
+  :  version:string
+  -> string
+  -> ( ((string * string list * (string * (string * string option) list) list)
        * Ast.NoLoc.TypeExpr.t)
-       Stdlib.Map.Make(Stdlib.StringLabels).t
-     * string
-     * int
+       list
+       * ((string * Ast.NoLoc.ModuleItem.type_param * Ast.NoLoc.annot)
+         * Ast.NoLoc.TypeExpr.t)
+         Stdlib.Map.Make(Stdlib.StringLabels).t
+       * string
+       * int
+       * Version.t
+     , [> `Empty_atd_file of string
+       | `Incoherent_version_field of string * string
+       | `No_version_field of string
+       ] )
+     result
```

Comment:
This begs for some type aliases to make it more readable.

Reply:
agreed!